src/featbin/concat-feats.cc

```diff
@@ -0,0 +1,96 @@
+// featbin/concat-feats.cc
+
+// Copyright 2013 Johns Hopkins University (Author: Daniel Povey)
+//           2015 Tom Ko
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include "base/kaldi-common.h"
+#include "util/common-utils.h"
+#include "matrix/kaldi-matrix.h"
+
+namespace kaldi {
+
+/*
+   This function concatenates several sets of feature vectors 
+   to form a longer set. The length of the output will be equal
+   to the sum of lengths of the inputs but the dimension will be
+   the same to the inputs.
+*/
+
+void ConcatFeats(const std::vector<Matrix<BaseFloat> > &in,
+                 Matrix<BaseFloat> *out) {
+  KALDI_ASSERT(in.size() >= 1);
+  int32 tot_len = in[0].NumRows(),
+      dim = in[0].NumCols();
+  for (int32 i = 1; i < in.size(); i++) {
+    KALDI_ASSERT(in[i].NumCols() == dim);
+    tot_len += in[i].NumRows();
+  }
+  out->Resize(tot_len, dim);
+  int32 len_offset = 0;
+  for (int32 i = 0; i < in.size(); i++) {
+    int32 this_len = in[i].NumRows();
+    out->Range(len_offset, this_len, 0, dim).CopyFromMat(
+        in[i]);
+    len_offset += this_len;
+  }
+}
+
+
+}
+
+int main(int argc, char *argv[]) {
+  try {
+    using namespace kaldi;
+    using namespace std;
+    
+    const char *usage =
+        "Concatenate feature files (assuming they have the same dimensions)\n"
+        "Usage: concat-feats <in-rxfilename1> <in-rxfilename2> [<in-rxfilename3> ...] <out-wxfilename>\n"
+        " e.g. concat-feats mfcc/foo.ark:12343 mfcc/foo.ark:56789 -\n"
+        "See also: copy-feats, copy-matrix, append-vector-to-feats, paste-feats\n";
```

Comment:
Please make sure that concat-feats appears as "See also:" in the usage message of paste-feats and copy-feats.